src/business-logic/services/rundown-timeline-service.ts

```diff
@@ -56,10 +73,11 @@ export class RundownTimelineService implements RundownService {
     const timeline: Timeline = this.timelineBuilder.getBaseTimeline()
 
     this.timelineRepository.saveTimeline(timeline)
-    await this.rundownRepository.saveRundown(rundown)
 
     const deactivateEvent: RundownEvent = this.rundownEventBuilder.buildDeactivateEvent(rundown)
     this.rundownEventEmitter.emitRundownEvent(deactivateEvent)
+
+    await this.rundownRepository.saveRundown(rundown)
   }
 
   public async takeNext(rundownId: string): Promise<void> {
```

Comment:
This method has had a growth spurt, to a point that i think it needs to be split into smaller pieces.
Splitting into 2-3 methods should be sufficent in cutting its size down. 

Much, if not all, of the newly added seems like it could be extracted to another method. 
Another candiate for code that could be extracted is the events that are build and fired at the end.

Extracting these two suggested pieces would cut down the size nicely, and the new methods would only take 1-2 arguements,

Reply:
Some effort has been put into reducing some of the duplication in this file